src/lib/krb5/os/dnssrv.c

```diff
@@ -45,11 +45,128 @@ krb5int_free_srv_dns_data (struct srv_dns_entry *p)
     }
 }
 
-/* Do DNS SRV query, return results in *answers.
+/*
+ * Prepares a DNS label of service.protocol.realm using a caller-supplied fixed
+ * buffer.  Since the protocol is not relevant to all callers it can be set to
+ * NULL to be excluded from the label.
+ */
```

Comment:
This should use the imperative mood, and can be more concise:

    Construct a DNS label of the form "service.[protocol.]realm.", placing the
    result into fixed_buf.  protocol may be NULL.